### ISSUE TITLE
Improve docs on how to restart CKAN

### ISSUE THREAD

**Issue by torfsen** [ckan/ckan#4512]
*Background: In stadt-karlsruhe/ckanext-extractor#14, a problem with background jobs was potentially caused by not restarting the background jobs worker after changes to the configuration or the CKAN virtualenv. This can happen easily, because the worker runs completely separately from the web server and many people just forget about it.*

The CKAN docs talk a lot about restarting CKAN, but the instructions are often inconsistent and/or lacking details (e.g. restarting the background jobs worker is usually not mentioned). I suggest a separate doc page that documents how to restart the various parts of CKAN's ecosystem and which is linked to from other pages instead of using ad-hoc descriptions of the process.

Here's a (probably incomplete) list of existing doc pages that mention restarting a production CKAN system:

- https://docs.ckan.org/en/2.8/maintaining/installing/install-from-package.html?highlight=restart
- https://docs.ckan.org/en/2.8/maintaining/upgrading/upgrade-package-to-minor-release.html?highlight=restart
- https://docs.ckan.org/en/2.8/maintaining/upgrading/upgrade-source.html?highlight=restart
- https://docs.ckan.org/en/2.8/maintaining/upgrading/upgrade-package-to-patch-release.html?highlight=restart
- https://docs.ckan.org/en/2.8/extensions/remote-config-update.html?highlight=restart
- https://docs.ckan.org/en/2.8/maintaining/upgrading/upgrade-postgres.html?highlight=restart
- https://docs.ckan.org/en/2.8/maintaining/filestore.html?highlight=restart
- https://docs.ckan.org/en/2.8/maintaining/email-notifications.html?highlight=restart
- https://docs.ckan.org/en/2.8/maintaining/tracking.html?highlight=restart
- https://docs.ckan.org/en/2.8/maintaining/getting-started.html?highlight=restart
- https://docs.ckan.org/en/2.8/maintaining/configuration.html?highlight=restart

**Comment by amercader** [comment]
@torfsen I'm not convinced that we can cover all situations in a single separate page. The exact "restarting"  commands can mean lots of different things depending on the setup and the environment.
Generally restarting CKAN means restarting the web server and any background jobs (core jobs, harvesters, etc). In the official docs this is Apache and Nginx, and the actual commands are listed (`sudo service restart X`), but people also run CKAN in gunicorn or uwsgi which will have different commands. Or use docker compose. On production, background jobs, harvesters etc are probably managed by supervisor so you need to do `supervisorctl restart all`.

Perhaps the issue is that the fact that the background jobs need to be restarted is not highlighted enough.
In any case I'm happy to be convinced otherwise so if you can propose a draft of how page would look like that would be great
Of course

**Comment by torfsen** [comment]
@amercader I totally agree that a single page that covers 100% of all cases is neither realistic nor desirable. However, I do think that a page that documents

* *what* needs to be done (e.g. "restart your webserver") and
* *how* that is done *in the standard setup* (as described in the other parts of the docs, i.e. Apache + Nginx + Supervisor)

is worth having. I'll try to prepare a draft once I find time for it.

**Comment by Nenad-N** [comment]
hi guys, 
We are trying to restart ckan, our installed version is 2.8.9, and it happens when we make changes to the configuration files, it does not save the changes. We tried restarting jetty8 and activating and deactivating our virtualenv the same problem happens to us.
